packages/providers/src/provider.ts

```diff
@@ -290,6 +290,13 @@ export default class Provider {
     return processGqlChain(chain);
   }
 
+  async getChainId(): Promise<number> {
+    const {
+      consensusParameters: { chainId },
+    } = await this.getChain();
+    return chainId.toNumber();
+  }
+
```

Comment:
Since people need to pass in the `chainId` when initializing a predicate, I created a helper method that makes it straight-forward for them to fetch it.